### PR TITLE
Dont merge! Pull request for Dell clients

### PR DESCRIPTION
**Pull request by DanPreda90**
I'm not sure what the issue is here, get a 401 response code saying that the access token and endpoint is incorrect even though I'm sure it isn't.

### CODE REVIEW

File: Backend/test.py [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+      }
+    }
+  ]
+}"""
+
+@app.route("/task")
 def hello_world():
-    return f"<p>{test_data}</p>"
-
+    h = {"Ocp-Acim-Subscription-Key":sys.argv[1]}
```
Comment (QuintenPohl) [inline Backend/test.py:44]: Try adding content type as well, e.g., `"Content-Type: application/json"`

File: Backend/test.py [hunk 2]
```diff
@@ -1,27 +1,49 @@
...
+    response = requests.post(url="https://dellsemanticproductlink.cognitiveservices.azure.com/language/analyze-text/jobs?api-version=2022-10-01-preview",headers=h,data=task_json)
```
Comment (QuintenPohl) [inline Backend/test.py:45]: Ensure the version is accesible

**Comment by phillips-sa** [comment]
Sidenote, I now see what you meant earlier about how the POST is long running https://learn.microsoft.com/en-us/azure/ai-services/language-service/concepts/use-asynchronously#submit-an-asynchronous-job-using-the-rest-api. Will be interesting to see how long it takes overall to get the final data back (POST request to trigger job + GET to retrieve results), something to consider if we're looking at optimizing overall performance down the line. 

Edit: 
Seeing in that same document they reference a client library they have for multiple languages including python, might be easier to try this out rather than manually building, sending and awaiting the requests (my understanding is the library would handle all that for you and return the result when its ready) : 
https://learn.microsoft.com/en-us/python/api/overview/azure/ai-textanalytics-readme?view=azure-python

**Comment by coraleaf0602** [comment]
> I'm not sure what the issue is here, get a 401 response code saying that the access token and endpoint is incorrect even though I'm sure it isn't.

I had the same issue in my code as well

**Comment by QuintenPohl** [comment]
> e issue is here, get a 401 response code saying that the access token and endpoint is incorrect even though I'm sure it isn't.

Is your environment variable setup correctly sys.argv[1]?

**Comment by coraleaf0602** [comment]
> Sidenote, I now see what you meant earlier about how the POST is long running https://learn.microsoft.com/en-us/azure/ai-services/language-service/concepts/use-asynchronously#submit-an-asynchronous-job-using-the-rest-api. Will be interesting to see how long it takes overall to get the final data back (POST request to trigger job + GET to retrieve results), something to consider if we're looking at optimizing overall performance down the line.
> 
> Edit: Seeing in that same document they reference a client library they have for multiple languages including python, might be easier to try this out rather than manually building, sending and awaiting the requests (my understanding is the library would handle all that for you and return the result when its ready) : https://learn.microsoft.com/en-us/python/api/overview/azure/ai-textanalytics-readme?view=azure-python

I found this link as well, but I wasn't sure if I should use it since that would kinda overwrite our previous progress. Will look into it now! Thanks Sarah

**Comment by phillips-sa** [comment]
> > Sidenote, I now see what you meant earlier about how the POST is long running https://learn.microsoft.com/en-us/azure/ai-services/language-service/concepts/use-asynchronously#submit-an-asynchronous-job-using-the-rest-api. Will be interesting to see how long it takes overall to get the final data back (POST request to trigger job + GET to retrieve results), something to consider if we're looking at optimizing overall performance down the line.
> > Edit: Seeing in that same document they reference a client library they have for multiple languages including python, might be easier to try this out rather than manually building, sending and awaiting the requests (my understanding is the library would handle all that for you and return the result when its ready) : https://learn.microsoft.com/en-us/python/api/overview/azure/ai-textanalytics-readme?view=azure-python
> 
> I found this link as well, but I wasn't sure if I should use it since that would kinda overwrite our previous progress. Will look into it now! Thanks Sarah

Yeah let me know how you get on with it, definitely save your previous work in a branch on here incase you decide to revert to the first way. Hopefully the package will make it so you don't need to worry about async, cuz with async its very easy to run into odd problems that are hard to troubleshoot 😄

**Pull request closed without merging**